contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3641,8 +3641,14 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 									char *permissions[] = {"select", "insert", "update", "references", "delete"};
 									for(i = 0; i < 5; i++)
 									{
-										if ((rol_spec->rolename != NULL) && !check_bbf_schema_for_entry(logical_schema, obj, permissions[i], rol_spec->rolename))
-											add_entry_to_bbf_schema(logical_schema, obj, permissions[i], rol_spec->rolename, obj_type);
+										if ((rol_spec->rolename != NULL) && (strcmp(rol_spec->rolename, "public") != 0))
```

Comment:
I think we can check for the following condition at the beginning of the function
```
if ((rol_spec->rolename != NULL) && (strcmp(rol_spec->rolename, "public") != 0))
```
Then, we can skip the same check for the rest of the function and assume that the condition is true.